File: app/src/main/java/b12app/vyom/com/bookmybus/view/Routes/RoutesPresenter.java
package b12app.vyom.com.bookmybus.view.Routes;

import android.content.Context;

import org.jetbrains.annotations.NotNull;

import b12app.vyom.com.bookmybus.model.JBusByRoute;
import b12app.vyom.com.bookmybus.data.remote.*;
import b12app.vyom.com.bookmybus.utils.VolleyHelper;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class RoutesPresenter implements RoutesContract.IPresenter,GetIDListener {

    private RoutesContract.IView iView;
    private Context context;
    private RoutesActivity routesActivity;
    private JApiService jApiService;


    public RoutesPresenter(RoutesActivity routesActivity) {
        this.routesActivity = routesActivity;
        this.context = routesActivity;
        this.iView = routesActivity;
        jApiService = JRetrofitInstance.getRetrofitInstance().create(JApiService.class);
//        new VolleyHelper(routesActivity).startRequest(1,1,1,1,this);
    }

    @Override
    public void getBusesForRoute(String id) {

        jApiService.getUser("1").enqueue(new Callback<JBusByRoute>() {
            @Override
            public void onResponse(Call<JBusByRoute> call, Response<JBusByRoute> response) {

                JBusByRoute jBusByRoute = response.body();
                iView.initRecyclerView(jBusByRoute.getBusinformation());

            }

            @Override
            public void onFailure(Call<JBusByRoute> call, Throwable t) {

            }
        });

    }

    @Override
    public void getId(@NotNull String id) {
        getBusesForRoute(id);
    }
}

File: app/src/main/java/b12app/vyom/com/bookmybus/view/routes/RoutesPresenter.kt
package b12app.vyom.com.bookmybus.view.routes

import android.content.Context

import b12app.vyom.com.bookmybus.model.JBusByRoute
import b12app.vyom.com.bookmybus.data.remote.*
import b12app.vyom.com.bookmybus.utils.VolleyHelper
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class RoutesPresenter(private val routesActivity: RoutesActivity) : RoutesContract.IPresenter, GetIDListener {

    private val iView: RoutesContract.IView
    private val context: Context
    private val jApiService: JApiService


    init {
        this.context = routesActivity
        this.iView = routesActivity
        jApiService = JRetrofitInstance.getRetrofitInstance().create(JApiService::class.java)

    }

    override fun getRouteId(startLatitude: String, startLongitude: String, endLatitude: String, endLongitude: String) {
        VolleyHelper(routesActivity).startRequest(startLatitude, startLongitude, endLatitude, endLongitude, this)
    }

    override fun getBusesForRoute(id: String) {

        jApiService.getUser(id).enqueue(object : Callback<JBusByRoute> {
            override fun onResponse(call: Call<JBusByRoute>, response: Response<JBusByRoute>) {

                val jBusByRoute = response.body()
                iView.initRecyclerView(jBusByRoute!!.businformation)

            }

            override fun onFailure(call: Call<JBusByRoute>, t: Throwable) {

            }
        })

    }

    override fun getId(id: String) {
        getBusesForRoute(id)
    }
}
